### PR TITLE
Cli safe mkdirs

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: cli/src/main/kotlin/commands/ScannerCommand.kt [hunk 1]
```diff
@@ -53,6 +53,7 @@ import org.ossreviewtoolkit.scanner.storages.SCAN_RESULTS_FILE_NAME
 import org.ossreviewtoolkit.utils.expandTilde
 import org.ossreviewtoolkit.utils.formatSizeInMib
 import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.safeMkdirs
```
Comment (nnobelis) [inline cli/src/main/kotlin/commands/ScannerCommand.kt:56]: alphabetical import ordering please.

**Review by nnobelis — CHANGES_REQUESTED** [review]

File: cli/src/main/kotlin/commands/ScannerCommand.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/ScannerCommand.kt]: Done.

**Review by nnobelis — APPROVED** [review]

**Review by MarcelBochtler — APPROVED** [review]

File: cli/src/main/kotlin/commands/AdvisorCommand.kt [hunk 2]
```diff
@@ -124,9 +124,10 @@ class AdvisorCommand : CliktCommand(name = "advise", help = "Run vulnerability d
 
         val ortResult = advisor.retrieveVulnerabilityInformation(input, skipExcluded).mergeLabels(labels)
 
+        outputDir.safeMkdirs()
```
Comment (fviernau) [inline cli/src/main/kotlin/commands/AdvisorCommand.kt:127]: Did this actually fix a bug?

**Review by fviernau — APPROVED** [review]

File: cli/src/main/kotlin/commands/AdvisorCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/AdvisorCommand.kt:127]: Yes. Well, I didn't verify it yet, but we *did* run into an

```
Writing scan result to '/output/results/scan-result.yml'.
Exception in thread "main" java.io.FileNotFoundException: /output/results/scan-result.yml (No such file or directory)
```

error.

Comment (fviernau) [reply cli/src/main/kotlin/commands/AdvisorCommand.kt:127]: Thanks!

**Pull request merged**